packages/contracts/contracts/L1/ValidatorPool.sol

```diff
@@ -193,26 +244,83 @@ contract ValidatorPool is ReentrancyGuardUpgradeable, Semver {
 
     /**
      * @notice Attempts to unbond corresponding to nextUnbondOutputIndex and returns whether the unbond was successful.
+     *         When unbound, it updates the next priority validator and sends a reward message to L2.
      *
      * @return Whether the bond has been successfully unbonded.
      */
     function _tryUnbond() private returns (bool) {
         uint256 outputIndex = nextUnbondOutputIndex;
-        address recipient = L2_ORACLE.getSubmitter(outputIndex);
 
         Types.Bond storage bond = bonds[outputIndex];
-        if (block.timestamp >= bond.expiresAt && bond.amount > 0) {
-            _increaseBalance(recipient, bond.amount);
-            nextUnbondOutputIndex = nextUnbondOutputIndex + 1;
-            emit Unbonded(outputIndex, recipient, bond.amount);
-            bond.amount = 0;
+        uint128 bondAmount = bond.amount;
+        if (block.timestamp >= bond.expiresAt && bondAmount > 0) {
```

Comment:
The bond.amount value is guaranteed to be positive from the time it is set, so the conditional here seems unnecessary.

```suggestion
        if (block.timestamp >= bond.expiresAt && bondAmount > 0) {
```
```suggestion
        if (block.timestamp >= bond.expiresAt) {
```

Reply:
If there is no bond, expiresAt is 0, which is always past the block.timestamp, so `bondAmount` is required.